rlberry/agents/torch/tests/test_ppo.py

```diff
@@ -197,5 +201,363 @@ def test_ppo():
     pporlberry_stats.clear_output_dir()
 
 
-if __name__ == "__main__":
-    test_ppo()
+def test_ppo_single_env():
+    env = gym_make("CartPole-v0")
+    agent = PPOAgent(
+        env,
+        learning_rate=1e-4,
+        optimizer_type="ADAM",
+    )
+    agent.fit(budget=1000)
+
+    saving_path = "rlberry/agents/torch/tests/agent_test_ppo_classic_env.pickle"
+
+    # VRemove previous save
+    if os.path.exists(saving_path):
+        os.remove(saving_path)
+    assert not os.path.exists(saving_path)
+
+    # test the save function
+    agent.save(saving_path)
+    assert os.path.exists(saving_path)
+
+    # test the loading function
+    test_load_env = gym_make("CartPole-v0")
+    loaded_agent = PPOAgent.load(saving_path, **dict(env=test_load_env))
+    assert loaded_agent
+
+    # test the agent
+    observation = test_load_env.reset()
+    for tt in range(50):
+        action = loaded_agent.policy(observation)
+        next_observation, reward, done, info = test_load_env.step(action)
+        if done:
+            next_observation = test_load_env.reset()
+        observation = next_observation
+
+    os.remove(saving_path)
+
+
+def test_ppo_multi_env():
+    env = gym_make("CartPole-v0")
+    agent = PPOAgent(env, learning_rate=1e-4, optimizer_type="ADAM", n_envs=3)
+    agent.fit(budget=1000)
+
+    saving_path = "rlberry/agents/torch/tests/agent_test_ppo_classic_env.pickle"
+
+    # VRemove previous save
+    if os.path.exists(saving_path):
+        os.remove(saving_path)
```

Comment:
Another way to do this is to use temporary folder. Python has a function for this. It is convenient because : unique name, and deleted automatically at reboot. In particular, this is convenient to not have litter files when running test locally. I think I made some of the rlberry tests with temporary folder (check out here: https://github.com/rlberry-py/rlberry/blob/7dfba03559258dfe512bee231f21ab95bb9db2af/rlberry/utils/check_agent.py#L181)

This comment is also applicable to the rest of the tests.